clash-lib/src/Clash/Util.hs

```diff
@@ -184,6 +186,25 @@ makeCachedU key l create = do
       l %= extendUniqMap key value
       return value
 
+-- | Cache the result of a monadic action using a 'OMap'
+makeCachedO
+  :: (MonadState s m, Uniquable k)
+  => k
+  -- ^ Key the action is associated with
+  -> Lens' s (OMap Unique v)
+  -- ^ Lens to the cache
+  -> m v
+  -- ^ Action to cache
+  -> m v
+makeCachedO key l create = do
+  cache <- use l
+  case OMap.lookup (getUnique key) cache of
+    Just value -> return value
+    Nothing -> do
+      value <- create
+      l %= (flip (OMap.|>)) (getUnique key, value)
```

Comment:
I'm not really a fan of qualifying operators, I'd rather see it in the import list and used "normally"

Reply:
Hmm yeah, I really don't like obscure operators like these and I generally just avoid them by using their named versions, but `OMap` doesn't export those. But I'll apply your suggestion anyway.

Reply:
Please keep the qualification, I’d confuse it with the `Data.Sequence` operator otherwise.

Reply:
2:1, sorry Alex